code/go/0chain.net/smartcontract/zcnsc/authorizer_test.go

```diff
@@ -0,0 +1,532 @@
+package zcnsc
+
+import (
+	"0chain.net/chaincore/chain"
+	"0chain.net/core/logging"
+	"go.uber.org/zap"
+	"math/rand"
+
+	//cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/chaincore/tokenpool"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+	"encoding/json"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+var (
+	stringEmpty = ""
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	chain.ServerChain = new(chain.Chain)
+	chain.ServerChain.Config = new(chain.Config)
+	chain.ServerChain.ClientSignatureScheme = "bls0chain"
+
+	logging.Logger = zap.NewNop()
+}
+
+func Test_TransferStateAfterAddingAuthorizer(t *testing.T) {
```

Comment:
Here and everywhere else: I think it is better style to have one test for each function and use subtests for different cases. 
`TestAddAuthorizer` with subtest "transfer state after adding authorizer".

This looks like you are testing 30 functions reather than two.

Reply:
@Sriep 
It would be, but to fit all tests into a single test, the **data** should be:
1. Homogeneous
2. Preconditions should be the same

**The logic flow** should be the same for all use cases.

I reviewed my tests again and saw that they cannot be converged into a single-use case, and the input data cannot be made homogeneous. Some tests look similar.